src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
+Animal.alive = [];
+
+Animal.checkIsAlive = () => {
+  for (const animal of Animal.alive) {
+    if (animal.health === 0) {
+      Animal.alive.splice(Animal.alive.indexOf(animal), 1);
+    }
+  }
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
```

Comment:
you can write this without constructor and supper just adding Animal.alive.push(this); in constructor in your class Animal and remove constructor from Hebrivore and add for Animal hidden === false

Reply:
agree, you can avoid using constructor here